packages/zoe/src/contractFacet/offerSafety.js

```diff
@@ -1,12 +1,14 @@
+// @ts-check
+
 /**
  * Helper to perform satisfiesWant and satisfiesGive. Is
  * allocationAmount greater than or equal to requiredAmount for every
  * keyword of giveOrWant?
- * @param {(Brand) => AmountMath} getAmountMath
- * @param {Proposal["give"] | Proposal["want"]} giveOrWant
+ * @param {(brand: Brand) => AmountMath} getAmountMath
+ * @param {ProposalRecord["give"] | ProposalRecord["want"]} giveOrWant
```

Comment:
(Typing only)

In a type declaration, what's the difference between
```js
ProposalRecord["give"]
```
vs
```js
ProposalRecord.give
```
? 
Why?

Would it make more sense to just declare
```js
@param {AmountKeywordRecord} giveOrWant
```
?
I would not have found it puzzling.

Reply:
> (Typing only)
> 
> In a type declaration, what's the difference between
> 
> ```js
> ProposalRecord["give"]
> ```
> 
> vs
> 
> ```js
> ProposalRecord.give
> ```

```
Cannot access 'ProposalRecord.give' because 'ProposalRecord' is a type, but not a namespace. Did you mean to retrieve the type of the property 'give' in 'ProposalRecord' with 'ProposalRecord["give"]'?ts(2713)
```

> Would it make more sense to just declare
> 
> ```js
> @param {AmountKeywordRecord} giveOrWant
> ```

Done.